examples/model-conversion/scripts/utils/semantic_check.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+
+import numpy as np
+import argparse
+import os
+import importlib
+
+from transformers import AutoTokenizer, AutoConfig, AutoModelForCausalLM, AutoModel
+from sklearn.metrics.pairwise import cosine_similarity
```

Comment:
We can avoid `scikit‑learn` dependency by implementing a `cosine_similarity` function.